marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/ingest/default/main.sjs

```diff
@@ -27,13 +23,25 @@ function main(id, content, options) {
     }
   }
 
-  let headers = {};
-
+  let headers = createHeaders(id, content, options);
   let envelope = datahub.flow.flowUtils.makeEnvelope(instance, headers, triples, outputFormat);
 
   return envelope;
 }
 
+function createHeaders(id, content, options) {
+  let headers = {};
+  for (let key in options.headers) {
```

Comment:
Is there a way we can functionize this to the hubUtils library for checking over the constants and returning the value so we just need to loop over the object and use the function to check and return the correct value?

Reply:
Sure. we can do that.